lib/plugins/aws/package/compile/events/alb/lib/validate.js

```diff
@@ -217,7 +218,21 @@ module.exports = {
   },
 
   validateAlbAuth(auth) {
-    auth.onUnauthenticatedRequest = auth.allowUnauthenticated ? 'allow' : 'deny';
+    const hasAllowUnauthenticated = auth.allowUnauthenticated !== undefined;
```

Comment:
Let's use more natural `!= null`

Reply:
I'm happy to change that but not sure if `!= null` makes it clearer. In this particular case, we only care about a case were the property is not defined not if the value is `null` `0` or any other.

Reply:
> we only care about a case were the property is not defined

If that would be the case, then proper check would be `'allowUnauthenticated' in auth`.

Still in JS it's not common to do such checks. Usually best practice is just to check whether a property holds some  value, where lack of value is represented by three possible scenarios:
- property not defined at all
- `undefined` assigned explicitly
- `null` assigned explicitly.

Having that, the most straightforward way to confirm property holds a value, is to follow with `obj.prop != null` (that excludes just three above cases and nothing more)